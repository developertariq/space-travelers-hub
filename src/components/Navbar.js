import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/Missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/MyProfile',
      text: 'My Profile',
    },
  ];
  return (
    <nav className="NavBar" data-testid="navbar">
      <ul className="MenuNav">
        {links.map((link) => (
          <li key={link.id}>
            <NavLink to={link.path} activeClassName="active-link" exact>
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
