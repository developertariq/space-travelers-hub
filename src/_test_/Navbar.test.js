import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the correct Rockets page', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const rocketsLink = getByText(/Rockets/i);
    fireEvent.click(rocketsLink);
    expect(window.location.pathname).toBe('/');
  });

  test('Rockets page is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const rocketsLink = getByText(/Rockets/i);
    fireEvent.click(rocketsLink);
    expect(rocketsLink.classList.contains('active')).toBe(true);
  });

  test('renders the correct Missions page', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const missionsLink = getByText(/Missions/i);
    fireEvent.click(missionsLink);
    expect(window.location.pathname).toBe('/Missions');
  });

  test('Missions page is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const missionsLink = getByText(/Missions/i);
    fireEvent.click(missionsLink);
    expect(missionsLink.classList.contains('active')).toBe(true);
  });

  test('renders the correct My Profile page', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const myProfileLink = getByText(/My Profile/i);
    fireEvent.click(myProfileLink);
    expect(window.location.pathname).toBe('/MyProfile');
  });

  test('My Profile page is active', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const myProfileLink = getByText(/My Profile/i);
    fireEvent.click(myProfileLink);
    expect(myProfileLink.classList.contains('active')).toBe(true);
  });
});
