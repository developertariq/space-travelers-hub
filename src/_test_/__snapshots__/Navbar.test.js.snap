// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="NavBar"
  data-testid="navbar"
>
  <ul
    className="MenuNav"
  >
    <li>
      <a
        activeClassName="active-link"
        aria-current="page"
        className="active"
        exact={true}
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        activeClassName="active-link"
        className=""
        exact={true}
        href="/Missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        activeClassName="active-link"
        className=""
        exact={true}
        href="/MyProfile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
